infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/pvc/CommonPVCStrategy.java

```diff
@@ -124,24 +146,24 @@ public void provision(KubernetesEnvironment k8sEnv, RuntimeIdentity identity)
       return;
     }
     log.debug("Provisioning PVC strategy for workspace '{}'", workspaceId);
+
     final Set<String> subPaths = new HashSet<>();
+    // Note that PVC name is used during prefixing
+    // It MUST be done before changing all PVCs references to common PVC
+    prefixVolumeMountsSubpaths(k8sEnv, identity.getWorkspaceId(), subPaths);
+
     String commonPVCName = getCommonPVCName(identity);
-    final PersistentVolumeClaim pvc = newPVC(commonPVCName, pvcAccessMode, pvcQuantity);
-    k8sEnv.getPersistentVolumeClaims().put(commonPVCName, pvc);
-    for (PodData pod : k8sEnv.getPodsData().values()) {
-      PodSpec podSpec = pod.getSpec();
-      List<Container> containers = new ArrayList<>();
-      containers.addAll(podSpec.getContainers());
-      containers.addAll(podSpec.getInitContainers());
-      for (Container container : containers) {
-        String machineName = Names.machineName(pod, container);
-        InternalMachineConfig machineConfig = k8sEnv.getMachines().get(machineName);
-        addMachineVolumes(
-            commonPVCName, workspaceId, subPaths, pod, container, machineConfig.getVolumes());
-      }
-    }
+    final PersistentVolumeClaim commonPVC = newPVC(commonPVCName, pvcAccessMode, pvcQuantity);
```

Comment:
An idea for code simplification - move new pvc creation and replacement to a separate method and make it return new pvc for reuse

Reply:
Move to private method `relacePVCsWithCommon`